easy_my_coop/models/subscription_request.py

```diff
@@ -750,26 +746,37 @@ def validate_subscription_request(self):
     @api.multi
     def block_subscription_request(self):
         self.ensure_one()
+        if self.state != "draft":
+            raise ValidationError(_("Only draft requests can be blocked."))
         self.write({"state": "block"})
 
     @api.multi
     def unblock_subscription_request(self):
         self.ensure_one()
+        if self.state != "block":
+            raise ValidationError(_("Only blocked requests can be unblocked."))
         self.write({"state": "draft"})
 
     @api.multi
     def cancel_subscription_request(self):
         self.ensure_one()
+        if self.state not in ("draft", "waiting", "done", "block"):
+            raise ValidationError(_("You cannot cancel a request in this " "state."))
```

Comment:
```suggestion
            raise ValidationError(_( "You cannot cancel a request in this "
                  "state."
             ))
```
Maybe you wanted to split line to fit 79 chars?